### PR TITLE
Initial setup of Javalin

### PR DESCRIPTION
**Pull request by ndoschek**
- Load model at request
- Use EMFJsonConverter
- First get request implemented: get model via /api/model/filepath (e.g. localhost:8081/api/model/Coffe.ecore)

### CODE REVIEW

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerModule.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *   SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ *******************************************************************************/
+package com.eclipsesource.modelserver.emf.launch;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Singleton;
+
+import io.javalin.Javalin;
+
+public class ModelServerModule extends AbstractModule {
```
Comment (tortmayr) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerModule.java]: Maybe we should combine the `ModelServerModule` and the `EMFModelServerModule`?
Or at least make the `EMFModelServerModule` a Subclass of `ModelServerModule`?

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import com.eclipsesource.modelserver.common.Routing;
+import com.google.inject.Inject;
+
+import io.javalin.Javalin;
+import io.javalin.JavalinEvent;
+
+public class ModelServerEntrypoint implements AppEntrypoint {
+	
+	private Javalin app;
+	private static final Logger logger = LoggerFactory.getLogger(ModelServerLauncher.class);
```
Comment (tortmayr) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java]: Maybe we should agree on common naming for the Logger instances.
I used LOG but I don't have a preference, I'd just like it to be consistent.

File: com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/AppEntrypoint.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ *   Licenses when the conditions for such availability set forth in the Eclipse
+ *   Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ *   with the GNU Classpath Exception which is available at
+ *   https://www.gnu.org/software/classpath/license.html.
+ *
+ *   SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ *******************************************************************************/
+package com.eclipsesource.modelserver.common;
+
+public interface AppEntrypoint {
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/AppEntrypoint.java:18]: Minor: casing, should read `AppEntryPoint` (here as well as anywhere else).

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerController.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import com.eclipsesource.modelserver.emf.EMFJsonConverter;
+import com.eclipsesource.modelserver.emf.ResourceManager;
+import com.eclipsesource.modelserver.emf.configuration.ServerConfiguration;
+import com.google.inject.Inject;
+
+import io.javalin.Context;
+import io.javalin.apibuilder.CrudHandler;
+
+public class ModelServerController implements CrudHandler {
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerController.java:34]: I'd rename this to `ModelController`

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	@SuppressWarnings("rawtypes")
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerEntrypoint.java:36]: No need for `rawtypes` here

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+public class ModelServerStartup {
+
+	@Inject(optional = true)
+	private Map<EntrypointType, AppEntrypoint> entrypoints = Collections.emptyMap();
+
+	public void boot(EntrypointType entrypointType, String[] args) {
+		Optional<AppEntrypoint> entryPoint = Optional.ofNullable(entrypoints.get(entrypointType));
+		entryPoint.orElseThrow(() -> new RuntimeException("Entrypoint not defined")).boot(args);;
+	}
+}
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java:35]: Newline missing

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import com.google.inject.Inject;
+
+public class ModelServerStartup {
+
+	@Inject(optional = true)
+	private Map<EntrypointType, AppEntrypoint> entrypoints = Collections.emptyMap();
+
+	public void boot(EntrypointType entrypointType, String[] args) {
+		Optional<AppEntrypoint> entryPoint = Optional.ofNullable(entrypoints.get(entrypointType));
+		entryPoint.orElseThrow(() -> new RuntimeException("Entrypoint not defined")).boot(args);;
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerStartup.java:33]: Unnecessary semicolon

**Review by tortmayr — COMMENTED** [review]
Looks good to me. Only minors/suggestions.

I think the mock ModelServer interface and EMFModelServer class can be removed now. Maybe the mechanism for caching already loaded models could be migrated to the ModelServerController

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerModule.java (hunk 1 above)
Comment (ndoschek) [reply com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/launch/ModelServerModule.java]: Yes, sounds good! Will work that in!

**Review by edgarmueller — APPROVED** [review]
Great work, thanks! I've pointed out only minor things, I'll probably fix them myself and push a PR.

**Pull request merged**